Dev.gzip prod

roedoejet opened the pull request:
We now deploy uncompressed files to GH Pages and pre-compressed data files to netlify. See [this comment](https://github.com/roedoejet/wordweaver/pull/73#issuecomment-1793096425) for more details.

roedoejet committed:
Revert "fix: only use gz in production"

This reverts commit cb91284ff1a059454ab4b26034f0f042a4c494b1.

roedoejet committed:
refactor: only use compression in certain envs

github pages cannot handle the appropriate response headers so we turn it off by default

we also deploy to netlify for serving pre-compressed gzip json

roedoejet committed:
refactor: move i18n data files to assets

roedoejet committed:
feat: add plausible analytics support

joanise commented on angular.json:
```diff
@@ -47,11 +52,56 @@
               ]
             },
-            "production": {
+            "ghpages": {
```
There's a lot of repetition between ghpages and netlify, I would prefer having a `prod` configuration and have `--configuration=ghpages,prod` and `--configuration=netlify,prod` in `package.json` where we call these, rather than duplicate everything. Not a big deal, though, not a required change.

joanise commented on projects/word-weaver/src/app/core/conjugation/conjugation.service.ts:
```diff
@@ -30,32 +25,16 @@ export class ConjugationService {
-        this.http.get<Conjugations>(settings.baseUrl + this.path, {
-          context: new HttpContext().set(SUPPRESS_ERROR, this.suppressError),
-        })
+        this.http.get<Conjugations>(settings.baseUrl + this.path)
```
So much simpler! :)

joanise reviewed (approved):
This is great. I have a minor optional change request to repeat ourselves less, but not necessary. 
I like how this PR means I can now have CI require up to date .json.gz files systematically, but they're only used in the deployment where they work.

When we do kawe, we'll need further tweaks because we'll only be able to commit the compressed files, but we'll only be able to spin up locally for testing if the uncompressed files are present. Maybe we can automate decompressing the compressed file when the uncompressed one is missing as part of the test deploy. E.g., `npm start` could trigger that before running `ng serve`.

roedoejet committed:
refactor: create production configuration for more dry configs

roedoejet committed:
feat(analytics): add opt-in and opt-out buttons for analytics

also update our i18n libraries since @biesbjerg is abandoned

fixes https://github.com/roedoejet/kawennonnis_ui/issues/3\#issuecomment-1799639236

this also required that we change from 'marker' to '_' for the new library to notice markers

this commit is missing french translations

joanise committed:
fix: add French translations for analytics messages

joanise committed:
refactor: use the full "marker" rather than the obscure "_"
